pkg/reconciler/taskrun/taskrun_test.go

```diff
@@ -4925,3 +4925,74 @@ status:
 		})
 	}
 }
+
+func TestReconcile_ReplacementsInStatusTaskSpec(t *testing.T) {
+	task := parse.MustParseTask(t, `
+metadata:
+  name: test-task-with-replacements
+  namespace: foo
+spec:
+  params:
+  - default: mydefault
+    name: myarg
+    type: string
+  steps:
+  - script: echo $(inputs.params.myarg)
```

Comment:
I'm only testing for replacement of a param here because that's good enough to prove the point, and is simplest to do. Note that this issue does _not_ show up with an inline `tr.Spec.TaskSpec`, interestingly.